transform/app/api/routes/solr_utils/solr.py

```diff
@@ -0,0 +1,65 @@
+from fastapi import APIRouter, Query
+from datetime import datetime
+import requests
+
+from app.transform.utils.loader import load_env_vars
+from app.transform.schemas.properties.env import (
+    SOLR_ADDRESS,
+    SOLR_PORT,
+    SOLR_COLLECTION_NAMES,
+)
+
+env_vars = load_env_vars()
+router = APIRouter()
+
+
+@router.post("/create_collections")
+async def create_solr_collections(
+    all_collection_config: str = Query(
+        ...,
+        description="Config name for 'all_collection'",
+        example="all_collection_oag56_v203",
+    ),
+    provider_config: str = Query(
+        ..., description="Config name for 'provider'", example="provider_v101"
+    ),
+    collection_prefix: str = Query(
+        "",
+        description="Prefix for collection names. "
+        "It is recommended to use convention 'oag<dump_version>'",
+        example="oag56",
+    ),
+    date: str = Query(
+        None,
+        description="Date string in the format 'YYYYMMDD'. "
+        "Defaults to the current date if not provided.",
+        example="20231114",
+    ),
+    num_shards: int = Query(1, description="Number of shards", example=1),
+    replication_factor: int = Query(1, description="Replication factor", example=1),
+):
+    """
+    Creates Solr collections for a singular data iteration.
+    """
+    date = date or datetime.now().strftime("%Y%m%d")
+
+    for collection in SOLR_COLLECTION_NAMES:
+        collection_name = f"{collection_prefix}_{date}_{collection}"
+        config = provider_config if collection == "provider" else all_collection_config
+
+        create_collection_url = (
+            f"{env_vars[SOLR_ADDRESS]}:{env_vars[SOLR_PORT]}/solr/admin/collections?action=CREATE"
+            f"&name={collection_name}&numShards={num_shards}&replicationFactor={replication_factor}"
+            f"&configName={config}&wt=xml"
+        )
+
+        response = requests.post(create_collection_url)
+
+        if response.status_code == 200:
+            print(f"Collection {collection_name} created successfully.")
```

Comment:
use logger instead of print. Here is logger.info

Reply:
You're right, I should include a logger.